src/QsCompiler/Transformations/Monomorphization.cs

```diff
@@ -167,7 +162,7 @@ public NamespaceTransformation(SyntaxTreeTransformation<TransformationState> par
                 public override QsCallable OnCallableDeclaration(QsCallable c)
                 {
                     var relevantAccessModifiers = this.SharedState.GetAccessModifiers.Apply(this.SharedState.TypeParams.Values)
-                        .Append(c.Access);
+                        .Append(c.Access).Append(Access.Internal); // monomorphized callables should never be public
```

Comment:
Couldn't this result in the internall modifier appearing more than once if the generic was already internal? Does that matter or is later code resilient to duplicate modifiers?

Reply:
This is already an enumerable collecting all access modifiers for types that are used. A couple of lines down .Min is used to determine the least permissible access. Hence, adding one to the bucket is fine matter.